### ISSUE TITLE
Problems installing on smf 2.0.11

### ISSUE THREAD

**Issue by bwaynef** [Spuds/SMF-HS4SMF#5]
I had some problems from the outset trying to install this mod.  When I go to upload it, I'm told there were some problems and to manually upload it via ftp client (i use cpanel's file manager instead) and eventually get access to the mod in the browse list.  From there I choose to install mod and I get errors dealing with Sources/Subs.php.

4.	Execute Modification	./Sources/Subs.php	Test failed
	1.	Add After	./Sources/Subs.php	Test failed

When I choose the notepad beside that I'm presented a window that says this:

> (Find) [Select]
> 	// Cache the output if it took some time...
>  (Add After) [Select]
> 	// HS4SMF, call the highslide function on this message, but not on signature, stats or when in the editor
> 	if((empty($cache_id) || substr($cache_id, 0, 3) != 'sig') && $context['current_action'] != 'stats' && $context['current_action'] != 'post' && $context['current_action'] != 'jseditor')
> 		hs4smf($message, $cache_id);

When I manually search the Subs.php file for "// Cache the output if it took some time..." it gives no results.  I see:

"// Cache these values..."
"// Cache these each load..."
"// Cache when we did it last."

Where do I go from here?

2.0.11 here.

**Comment by Spuds** [comment]
The first problem is nothing to do with the mod, its a permissions issue.  If the user that the web server runs under does not have write permissions to the package directory, then you have to use FTP.

The second issue, I'd guess you installed another mod that messed up that line.  Looking at 2.0.11 code (fresh download), Subs.php, line 2420 is ```	// Cache the output if it took some time...``` so the line is there as defined.

**Comment by bwaynef** [comment]
Thanks.  I'll look around that line to see if I can manually change it.

**Comment by bwaynef** [comment]
Strange, I didn't see this before.  Line 2446 says "// Cache the output if it took some time".  Right before that I see "// End Auto Embed Media Pro".  I guess the autoembed mod changed it.  I'll see what needs to be done and stick it in there.  (I need to figure out why the pkg manager didn't see this.)

**Comment by bwaynef** [comment]
I went ahead with the install, made the manual change to Subs.php as mentioned above, ...and I see no effect when I view pictures.

**Comment by bwaynef** [comment]
After applying the changes w/ the package manager & manually, when I load an image it simply loads a new window with the image.  It no longer loads the picture in-thread, but there's no integration w/ the rest of the forum.  It just loads the picture upper-left of a white page.

**Comment by bwaynef** [comment]
I also verified its not just the theme I'm using.  It doesn't work on either of the themes I have installed.

**Comment by Spuds** [comment]
Does it work on the default theme?

**Comment by bwaynef** [comment]
It didn't work on any theme I tried it on.  I ended up uninstalling the mod and went a different route, though I'd like to try it again at some point.

For some reason, as I mentioned above, it didn't recognize the strings in the file for the insert after/befores.  The first change in Subs.php the test failed.  I found the string manually.  The second string passed, but when I looked at the file the change to it wasn't applied.

I also checked permissions and they were the same on that file as one other files that passed.
